File: examples/breakout_ioexpander/button/button.cpp
#include "pico/stdlib.h"
#include <stdio.h>

#include "breakout_ioexpander.hpp"

using namespace pimoroni;

//Connect a button between this pin and ground
static const uint8_t IOE_BUTTON_PIN = 14;

BreakoutIOExpander ioe(0x18);
bool last_state = true;

int main() {
#ifdef PICO_DEFAULT_LED_PIN
  gpio_init(PICO_DEFAULT_LED_PIN);
  gpio_set_dir(PICO_DEFAULT_LED_PIN, GPIO_OUT);
#endif

  stdio_init_all();

  if(ioe.init()) {
    printf("IOExpander found...\n");

    ioe.set_mode(IOE_BUTTON_PIN, IOExpander::PIN_IN_PULL_UP);

    while(true) {
      bool state = ioe.input(IOE_BUTTON_PIN);
      if(state != last_state) {
        if(state) {
          printf("Button has been released\n");
#ifdef PICO_DEFAULT_LED_PIN
          gpio_put(PICO_DEFAULT_LED_PIN, false);
#endif
        }
        else {
          printf("Button has been pressed\n");
#ifdef PICO_DEFAULT_LED_PIN
          gpio_put(PICO_DEFAULT_LED_PIN, true);
#endif
        }

        last_state = state;
      }

      sleep_ms(20);
    }
  }
  else {
    printf("IOExpander not found :'(\n");
#ifdef PICO_DEFAULT_LED_PIN
    gpio_put(PICO_DEFAULT_LED_PIN, true);
#endif
  }

  return 0;
}

File: micropython/examples/breakout_ioexpander/button.py
import time
from pimoroni_i2c import PimoroniI2C
from breakout_ioexpander import BreakoutIOExpander

PINS_BREAKOUT_GARDEN = {"sda": 4, "scl": 5}
PINS_PICO_EXPLORER = {"sda": 20, "scl": 21}

ioe_button_pin = 14

i2c = PimoroniI2C(**PINS_BREAKOUT_GARDEN)
ioe = BreakoutIOExpander(i2c, address=0x18)

ioe.set_mode(ioe_button_pin, BreakoutIOExpander.PIN_IN_PU)

last_state = True

while True:
    state = ioe.input(ioe_button_pin)
    if state is not last_state:
        if state:
            print("Button has been released")
        else:
            print("Button has been pressed")
        last_state = state

    time.sleep(0.02)
